### PR TITLE
Remove dead code

### PR DESCRIPTION
**Pull request by dmage**

### CODE REVIEW

File: data/model/user.py [hunk 1]
```diff
@@ -363,17 +363,6 @@ def get_or_create_robot_metadata(robot):
...
 
-def update_robot_metadata(robot, description="", unstructured_json=None):
-    """
-    Updates the description and user-specified unstructured metadata associated with a robot account
-    to that specified.
-    """
-    metadata = get_or_create_robot_metadata(robot)
-    metadata.description = description
-    metadata.unstructured_json = unstructured_json or metadata.unstructured_json or {}
-    metadata.save()
```
[the remark this answers is not in the document]
Comment (dmage) [reply data/model/user.py:374]:
```
quay (master)$ grep --exclude-dir=.git update_robot_metadata -r .
./data/model/user.py:def update_robot_metadata(robot, description="", unstructured_json=None):
quay (master)$
```

It is never explicitly called.  `create_robot` creates metadata directly:

https://github.com/quay/quay/blob/f28b35cc311dae94671cc6bce4b11abc9a68e917/data/model/user.py#L350-L354

[the remark this answers is not in the document]
Comment (ibazulic) [reply data/model/user.py:374]: Ah, okay, I thought we can add descriptions even after a robot account is created. Which would be expected.

**Comment by ibazulic** [comment]
> Is there a reason for removing a bunch of tests?

@kleesc aren't those from the old validator? the one we had before config tool was invented?

**Comment by dmage** [comment]
@kleesc there are functions and classes that are used only by tests, and these tests verify only these functions and classes. For example, FastIndexList:

https://github.com/quay/quay/blob/f28b35cc311dae94671cc6bce4b11abc9a68e917/util/morecollections.py#L15-L19

It is used only by its tests:

https://github.com/quay/quay/blob/f28b35cc311dae94671cc6bce4b11abc9a68e917/util/test/test_morecollections.py#L44-L58

Or validators. I haven't found what uses them, so I removed them with all their tests.

So far I found only one thing that dynamically discovers classes that otherwise aren't used:

https://github.com/quay/quay/blob/f28b35cc311dae94671cc6bce4b11abc9a68e917/notifications/notificationevent.py#L84-L92

**Pull request closed without merging**